src/indexer/json_term_writer.rs

```diff
@@ -325,7 +327,7 @@ impl<'a> JsonTermWriter<'a> {
         self.close_path_and_set_type(T::to_type());
         self.term_buffer
             .as_mut()
-            .extend_from_slice(val.to_u64().to_be_bytes().as_slice());
+            .extend_from_slice(val.to_u64(None).to_be_bytes().as_slice());
```

Comment:
@fulmicoton I am not applying the precision here to DateTime. my reasoning is that this is all indexing though we are pushing fast values. Am I wrong?

Reply:
Great question!

`FastValue` is a poorly named trait. 
It is not only used for fast values, it is also used for the inverted index.

In reality it offers a strictly monotonic mapping from the type to U64 (and its reciprocal). 

We could be named `MonoMapU64` or something like that.
Can you fill a ticket to rename it?

Reply:
Back to your question... in the `json_term_writer` we should apply some truncation on dates.

I suggest we always truncate with second precision?

We should probably truncate to seconds always for regular (=not Json) date fields too; and only use the precision setting for fast fields.

Besides this is the current behavior of tantivy in a sense.